homeassistant/components/verisure.py

```diff
@@ -0,0 +1,89 @@
+"""
+components.verisure
+~~~~~~~~~~~~~~~~~~
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP,
+    CONF_USERNAME, CONF_PASSWORD)
+
+DOMAIN = "verisure"
+DEPENDENCIES = []
+REQUIREMENTS = [
+    'https://github.com/persandstrom/python-verisure/archive/master.zip'
+    ]
+
+_LOGGER = logging.getLogger(__name__)
+
+MY_PAGES = None
+STATUS = {}
+
+MIN_TIME_BETWEEN_REQUESTS = timedelta(seconds=5)
+
+
+def setup(hass, config):
+    """ Setup the Verisure component. """
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_USERNAME, CONF_PASSWORD]},
+                           _LOGGER):
+        return False
+
+    from verisure import MyPages
+
+    STATUS[MyPages.DEVICE_ALARM] = {}
+    STATUS[MyPages.DEVICE_CLIMATE] = {}
+    STATUS[MyPages.DEVICE_SMARTPLUG] = {}
+
+    global MY_PAGES
+    MY_PAGES = MyPages(
+        config[DOMAIN][CONF_USERNAME],
+        config[DOMAIN][CONF_PASSWORD])
+    MY_PAGES.login()
+    update()
+
+    def stop_verisure(event):
+        """ Stop the Verisure service. """
+        MY_PAGES.logout()
+
+    def start_verisure(event):
+        """ Start the Verisure service. """
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_verisure)
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_verisure)
```

Comment:
I think we can even remove all of it. Wouldn't the session just expire at some point if we don't logout?

If we do keep it here I actually like only listening for STOP after START as it makes logically more sense.